server/courselist.rb

```diff
@@ -259,7 +267,8 @@ def self.configureStatic
 
     # read in yml for the build configuration into a class variable
     begin
-      config_hash[:build] = self.get_local_config_yml(config_hash[:build_file], "./server/local/build.yml")
+      config_hash[:build] = self.get_local_config_yml(config_hash[:build_file], "./server/local/build.yml",false)
+      logger.info "build.yml file is optional"
```

Comment:
So build.yml file is optional, so that the get_local_config_yml would not throw fatal error when neither config_hash[:build_file] or ./server/local/build.yml file could be found. However, config_hash[:build] would be empty or nil at this time, will that be problem for line284?